ArraysHomework/ArraysHomework.cs

```diff
@@ -11,42 +12,49 @@ public void Arrays()
         {
             // Create an array of ints of size 100
             int[] myArray = new int[100];
+            
             Assert.IsTrue(myArray.Length == 100);
 
-            // take the array made above and fill it so these tests pass  
+            // take the array made above and fill it so these tests pass
+
             myArray[0] = 100;
             myArray[50] = 150;
             myArray[68] = 168;
+
             Assert.IsTrue(myArray[0] == 100);
             Assert.IsTrue(myArray[50] == 150);
             Assert.IsTrue(myArray[68] == 168);
 
             // fill all of the values in to make the loop on line 28 pass as is.
             // Do not edit the loop that assert the passing case. Instead, add code above it to make it pass.
-
+            int sum = 0; 
+            for (int i = 0; i < myArray.Length; i++)
+            {        
+                myArray[i] = sum;
+                sum++;
+            }
             // hint you need some form of repetition.
+
             for (int i = 0; i < myArray.Length; i++)        // Don't edit from HERE
             {                                               //
                 myArray[i] = 10;                            //
                 Assert.IsTrue(myArray[i] == 10);            // to HERE
 
             }
 
-
             // what wrong with this array declaration? Fix it and make it compile
-            string stringArray = new string[5];
+            string[] stringArray = new string[5];
 
             // What about this one? Fix it and make it compile
-            bool[] boolArray = new bool[];
+            bool[] boolArray = new bool[10];
 
             // What about this one? Fix it and make it compile
-            bool[] boolArray2 = new [20];
+            bool[] boolArray2 = new bool[20];
 
             // Take this array and copy its contents into a new array 
             char[] copyMe = { 'a', 'b', 'c' };
             char[] copiedArray = new char[3];
-
-            
+            copiedArray = copyMe;
```

Comment:
We haven't talked much about references yet, but this isn't making a copy. It's setting the copiedArray variable to point to the same array that copyMe points to - they become different names for the same thing in memory.
Don't use `copiedArray = copyme`. Instead, assign each of the individual values inside of `copiedArray`.